### ISSUE TITLE
Crash of the app when we mix flat json keys, and keys that are the prefix.

### ISSUE THREAD

**Issue by ruano-a** [intlify/vue-i18n-next#1365]
### Reporting a bug?

When using the option "flatJson: true", if my messages to translate are for instance the following :

`
{
    'animal.dog': 'Dog',
    'animal': 'Animal'
}
`

Once you start the app, the page is white and in console you have the message:
`
vue-i18n.esm-bundler.js:164 Uncaught TypeError: Cannot create property 'dog' on string 'Dog'
    at handleFlatJson (vue-i18n.esm-bundler.js:164:42)
    at getLocaleMessages (vue-i18n.esm-bundler.js:204:17)
    at createComposer (vue-i18n.esm-bundler.js:325:27)
    at createVueI18n (vue-i18n.esm-bundler.js:909:26)
    at vue-i18n.esm-bundler.js:2294:31
    at EffectScope.run (reactivity.esm-bundler.js:37:24)
    at createGlobal (vue-i18n.esm-bundler.js:2294:21)
    at createI18n (vue-i18n.esm-bundler.js:2122:37)
    at main.js?t=1680611143315:47:14
`

Of course this is fixable by using instead:

`
{
    'animal.dog': 'Dog',
    'animal.label': 'Animal'
}
`

But it's still a bug.

Sometimes it doesn't crash the whole thing, but it still causes one of the keys not to be found. Extremely confusing error when you're not aware of the bug.

### Expected behavior

The translation (and the app) should still work if you mix flat json keys (with a dot in it) and keys that are the prefix.

### Reproduction

Just do what I said in the explaination.

It also happens in cases like :

`
{
    'animal.stuff.dog': 'Dog',
    'animal.stuff': 'Animal'
}
`

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 AMD Ryzen 7 5800H with Radeon Graphics
    Memory: 3.85 GB / 15.36 GB
  Binaries:
    Node: 16.14.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.18 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.5.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1413.0), Chromium (111.0.1661.62)
    Internet Explorer: 11.0.22621.1
  npmPackages:
    @vitejs/plugin-vue: ^4.0.0 => 4.1.0
    vite: ^4.1.4 => 4.2.1
    vue: ^3.2.47 => 3.2.47
    vue-i18n: ^9.2.2 => 9.2.2
    vue-router: ^4.1.6 => 4.1.6
    vuetify: ^3.1.12 => 3.1.12
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/intlify/vue-i18n-next/blob/master/.github/CONTRIBUTING.md)
- [X] Read the [Documentation](https://vue-i18n.intlify.dev/)
- [X] Check that there isn't [already an issue](https://github.com/intlify/vue-i18n-next/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussions](https://github.com/intlify/vue-i18n-next/discussions)

**Comment by kazupon** [comment]
Thank you for your reporting!
Sorry for my late reply. 🙇 

Unfortunately, I could not reproduce this issue.
We need minimal reproduction, so Could you give us your it please? 🙏

**Comment by ruano-a** [comment]
> Thank you for your reporting! Sorry for my late reply. 🙇
> 
> Unfortunately, I could not reproduce this issue. We need minimal reproduction, so Could you give us your it please? 🙏

This could help : https://codesandbox.io/s/brave-wilson-rr7psn?file=/src/App.vue

**Comment by kazupon** [comment]
Thanks!
I could see your issue with my eyes. 👀 
I'll fix this issue for vue-i18n v9.3.0
